ocaml/testsuite/tests/typing-layouts-word/alloc.ml

```diff
@@ -98,3 +99,12 @@ end
 
 let () = Collatz_unboxed.go ()
 let () = Collatz_boxed.go ()
+
+let[@inline never] literal_test x y =
+  let open Nativeint_u in
+  let[@inline never] f x y = (#1n + x) * (y - #4n) in
+  match x with
+  | #2n ->  (f x y) / (#3n % #10n)
+  | _ -> #0n
+
+let _ = measure_alloc "literals (should be -1): %nd" (fun () -> literal_test #2n #3n)
```

Comment:
It would be good to add a test (somewhere) using a literal that's out-of-bounds of a normal int63.